sharktank/sharktank/layers/paged_attention.py

```diff
@@ -283,21 +288,32 @@ def read(
 
         # Gather both partitions and split post gather. This is more
         # computationally efficient without gather fusion:
-        subblock_table = page_table.flatten(start_dim=0, end_dim=1)
+        page_table = page_table.flatten(start_dim=0, end_dim=1)
         page_stride = self.pipeline_to_block_count[pipeline]
 
         transformer_block_index = torch.full(
             (bs, block_seq_len), transformer_block_index, device=self.device
         )
         subblock_ids = page_ids * page_stride + transformer_block_index
-        selected = ops.index_select(subblock_table, 0, subblock_ids.flatten(0, 1))
+        selected = ops.index_select(page_table, 0, subblock_ids.flatten(0, 1))
 
         selected = selected.unflatten(0, blocked_shape[:2])
         key = selected[:, :, 0, :].flatten(1, 2)
         value = selected[:, :, 1, :].flatten(1, 2)
 
         return key, value
 
+    @staticmethod
+    def deshard_like(like, arg):
+        devices = like.devices
+
+        if isinstance(arg, SplitPrimitiveTensor):
+            assert all([a == b for a, b in zip(devices, arg.devices)])
+
+        assert len(set(arg.devices)) == len(arg.devices)
```

Comment:
Rereading this, I don't know when we would want to support duplicate devices, these are logical devices not physical ones.

This needs a similar assert but for `devices` as the current implementation will silently fail when `like` has duplicate devices.